src/radical/pilot/agent/scheduler/base.py

```diff
@@ -448,15 +448,13 @@ def _change_slot_states(self, slots, new_state):
                 raise RuntimeError('inconsistent node information')
 
             # iterate over cores/gpus in the slot, and update state
-            cores = rank['core_map']
-            for cslot in cores:
-                for core in cslot:
+            slot_ranks = rank['core_map']
```

Comment:
I kinda felt that this name fits - by default each slot represents one rank, but in case of shared gpu(s) slot will represent multiple ranks, i.e. ranks per slot